src/components/scenes/GameOverSingleplayer.js

```diff
@@ -0,0 +1,78 @@
+import { Text, View } from "react-native";
+import { useEffect, useContext } from "react";
+import { transact, tx } from "@instantdb/react-native";
+
+import { UserContext } from "@/Context";
+import SafeView from "@/components/shared/SafeView";
+import {
+  RegularButton,
+  primaryBackgroundColor as bgColor,
+  infoTextColor as textColor,
+} from "@/components/shared/styles";
+
+// Consts
+// ------------------
+
+// Screen
+// ------------------
+function GameOverSingleplayer({ navigation, route }) {
+  const user = useContext(UserContext);
+  const { score } = route.params;
+  const { id: userId, highScore } = user;
+
+  useEffect(() => {
+    if (score > highScore) {
+      transact(tx.users[userId].update({ highScore: score }));
+    }
+  }, []);
```

Comment:
@stopachka  Curious, do you think it's no harm to add it in? My intent was for this to run once and only once on mount.

Reply:
i.m.o there could be harm. Mainly, it's harder to actually test the "GameOver but somehow data changed" state. Because of that, we wouldn't _really_ know that useEffect will work in those cases anyways. 

Hence why good to keep as is for now